distributed/scheduler.py

```diff
@@ -8073,7 +8099,9 @@ async def check_worker_ttl(self):
                     self.worker_ttl,
                     ws,
                 )
-                await self.remove_worker(address=ws._address)
+                await self.handle_remove_worker(
+                    address=ws._address, stimulus_id=f"check-worker-ttl-{time()}"
+                )
```

Comment:
Pehaps check_worker_ttl should be decorated with `expects_stimulus`? I guess an initialising stimulus could be provided when setting up the PeriodicCallback